### ISSUE TITLE
[BUG] Cannot find module 'telefunc/react-streaming' or its corresponding type declarations.

### ISSUE THREAD

**Issue by Maxvien** [brillout/telefunc#76]
Hi @brillout,

I find this bug when try to import `telefunc/react-streaming`.

```ts
import { useData } from 'telefunc/react-streaming';
```

Error: `Cannot find module 'telefunc/react-streaming' or its corresponding type declarations.`

By the way, `vite`, `vite-plugin-ssr`, `telefunc` and `react-streaming` are awesome. I really like them ❤

**Comment by brillout** [comment]
The example at `/examples/react-streaming/` is working and uses that import, so I'm not sure why it doesn't work for you.

I'll need a reproduction.



> By the way, `vite`, `vite-plugin-ssr`, `telefunc` and `react-streaming` are awesome. I really like them ❤

I'm glad you like them. In case you're able to convince your employer: https://github.com/sponsors/brillout &mdash; it makes a big difference!

**Comment by Maxvien** [comment]
Hello @brillout,

I noticed that the react-streaming package is missing in the node_modules/telefunc directory, which could be the cause of the issue.

Here is a screenshot showing the missing package:

![image](https://github.com/brillout/telefunc/assets/1142799/beb35f73-b77e-45ed-ab87-06470909a5ac)

I am currently using your libraries to build my stack, and if they work perfectly, I would be happy to promote them within the JavaScript community in Vietnam.

Thank you!

**Comment by brillout** [comment]
> I noticed that the `react-streaming` is missing in the `node_modules/telefunc` directory, which could be the cause of the issue.

It's a peer dependency: make sure to install it.

> I am currently using your libraries to build my stack, and if they work perfectly, I would be happy to promote them within the JavaScript community in Vietnam.

💯

**Comment by Maxvien** [comment]
I can assure you that I have the react-streaming package installed.

Currently, I have to import the `useData` hook lie this:

```
import { useData } from 'telefunc/dist/esm/shared/react-streaming/useData';
```

**Comment by brillout** [comment]
I'll need a reproduction.

**Comment by Maxvien** [comment]
## Environment
Node: 18.16.1
NPM: 9.5.1
OS: Ubuntu 22.04

## Example
https://github.com/brillout/telefunc/tree/main/examples/react-streaming

## Dependencies
I must install dependencies  with the command below, because there is a problem: `Fix the upstream dependency conflict, or retry this command with --force or --legacy-peer-deps to accept an incorrect (and potentially broken) dependency resolution.`

```
npm i --force
```
## Status
The bug is also in your example:

![image](https://github.com/brillout/telefunc/assets/1142799/f679c511-00f4-40bc-b39b-4bd5ecf1efa4)

**Comment by brillout** [comment]
Fix released in `0.1.58`.

**Comment by Maxvien** [comment]
Thank you!
